middlewares/osio/osio.go

```diff
@@ -74,7 +77,30 @@ type OSIOAuth struct {
 	RequestSrvAccToken    SrvAccTokenLocator
 	RequestSecretLocation SecretLocator
 	RequestTokenType      TokenTypeLocator
-	cache                 *Cache
+	nsCache               *Cache
+	tokenCache            *Cache
+}
+
+type paramMap map[string]string
+
+func (m paramMap) getKey(delim string) string {
+	key := ""
+	if m[NamespaceType] != "" {
+		key = key + delim + m[NamespaceType]
+	}
+	if m["space"] != "" {
+		key = key + delim + m["space"]
+	}
+	return key
+}
+
+type httpErr struct {
+	err        string
+	statusCode int
+}
```

Comment:
HTTP error returned from tenant_service will be store with this special HTTP error type.